synapse/handlers/federation.py

```diff
@@ -1707,6 +1714,13 @@ def _handle_new_event(
                     self.store.remove_push_actions_from_staging, event.event_id
                 )
 
+        # If there's an expiry timestamp, schedule the redaction of the event.
+        expiry_ts = event.content.get(EventContentFields.SELF_DESTRUCT_AFTER)
```

Comment:
there are other places where events get persisted which I don't think you've caught (on backfill, maybe?). Can we push this down into `self.persist_events_and_notify` ?